cloud/aws/services/ec2/eips.go

```diff
@@ -30,11 +47,25 @@ func (s *Service) allocateAddress(clusterName string) (string, error) {
 		return "", errors.Wrapf(err, "failed to create Elastic IP address")
 	}
 
-	if err := s.createTags(clusterName, *out.AllocationId, ResourceLifecycleOwned, nil); err != nil {
-		return "", errors.Wrapf(err, "failed to tag elastic IP %q", *out.AllocationId)
+	name := fmt.Sprintf("%s-eip-%s", clusterName, role)
+	if err := s.createTags(clusterName, *out.AllocationId, ResourceLifecycleOwned, name, role, nil); err != nil {
+		return "", errors.Wrapf(err, "failed to tag elastic IP %q", aws.StringValue(out.AllocationId))
 	}
 
-	return *out.AllocationId, nil
+	return aws.StringValue(out.AllocationId), nil
+}
+
+func (s *Service) describeAddresses(clusterName string, role string) ([]*ec2.Address, error) {
+	out, err := s.EC2.DescribeAddresses(&ec2.DescribeAddressesInput{
+		Filters: []*ec2.Filter{
+			s.filterCluster(clusterName),
+			s.filterAWSProviderRole(role),
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to describe elastic IPs %q", err)
```

Comment:
This should just be `errors.Wrap(err, "failed to describe elastic IPs")`

Reply:
but upon further inspection, you can instead return `s.EC2.DescribeAddresses(...` directly